roles/cnf_cert/tasks/pre-run.yml

```diff
@@ -237,6 +209,17 @@
   - name: Create variable with tnf commit SHA
     set_fact:
       tnf_commit_sha: "{{ tnf_commit_sha_cmd.stdout }}"
+
+  - name: Create cnf-certification-test component calling to include_components role
+    vars:
+      gits_list:
+        - "{{ tnf_dir }}"
+      ic_gits: "{{ gits_list|flatten }}"
+      ic_rpms: []
+      ic_dev_gits: []
```

Comment:
Just curious, I assume the play `track_dev_git_repo.yml` will never be called from CNF role tasks if we pass this `ic_dev_gits: []` right? it will always ended up using `track_git_repo.yml` because we pass the `ic_gits` list with the `tnf_dir`. So this is expected right?

Also, in all the jobs you referenced, I see that the play `track_dev_git_repo.yml` is the agent pre-run, creates the git components, I just couldn't find where we pass `ic_dev_gits`, by default is empty, so please if you happen to know where is that overridden, I'll appreciate it : )

Reply:
Hi @manurodriguez ! thanks for your comments, I'll answer you below:

> Just curious, I assume the play track_dev_git_repo.yml will never be called from CNF role tasks if we pass this ic_dev_gits: [] right? it will always ended up using track_git_repo.yml because we pass the ic_gits list with the tnf_dir. So this is expected right?

Exactly, that's expected. The thing is, before, the cnf-certification-test component when testing stable versions (so, when the code doesn't come from a PR; for example, `v4.5.2`, `HEAD`, etc.) was created automatically, but now it isn't. With this code, we use `include_components` to create that component. As the code doesn't come from a change, we need to use `ic_gits`, and the rest of arguments are passed as empty list to avoid calling them.

> Also, in all the jobs you referenced, I see that the play track_dev_git_repo.yml is the agent pre-run, creates the git components, I just couldn't find where we pass ic_dev_gits, by default is empty, so please if you happen to know where is that overridden, I'll appreciate it : )

It's updated in the test-runner: https://github.com/redhat-cip/dci-pipeline/blob/master/tools/test-runner#L286-L289; so, for each folder extracted by extract-dependencies (so, they come from a change), they're added to `dev_gits_to_components` variable, which is eventually passed to `include_components role through `ic_dev_gits`. That's the way it's working for ansible-collection-redhatci-ocp and cnf-certification-test (where we are testing a PR).

Reply:
Thanks for the clarification, now I see where the magic happens : )